### PR TITLE
Update swarm chequebook for latest Solidity

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
I think the bytecode in the .go file needs to be updated before merging this.

File: contracts/chequebook/contract/chequebook.sol [hunk 1]
```diff
@@ -21,26 +23,23 @@ contract chequebook is mortal {
         uint8 sig_v, bytes32 sig_r, bytes32 sig_s) {
         // Check if the cheque is old.
         // Only cheques that are more recent than the last cashed one are considered.
-        if(amount <= sent[beneficiary]) return;
+        require(amount > sent[beneficiary]);
         // Check the digital signature of the cheque.
-        bytes32 hash = sha3(address(this), beneficiary, amount);
-        if(owner != ecrecover(hash, sig_v, sig_r, sig_s)) return;
+        bytes32 hash = keccak256(address(this), beneficiary, amount);
+        require(owner == ecrecover(hash, sig_v, sig_r, sig_s));
```
Comment (axic) [inline contracts/chequebook/contract/chequebook.sol]: These two `require` changes change behaviour: before if the owner/amount didn't match, it resulted in a successful execution without doing anything; now it results in a failed execution using the `revert` opcode (remainder gas is not consumed).

Comment (holiman) [reply contracts/chequebook/contract/chequebook.sol]: Is that good or bad @nagydani ?

Comment (nagydani) [reply contracts/chequebook/contract/chequebook.sol]: I think, it is okay. The actual effect is the same: nothing changes in the state and remaining gas is not consumed.

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 94fb8b186e5a7e60797b4bf8c61de51eecf5ee75
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Pull request merged**